lib/resources/order-recipients.js

```diff
@@ -0,0 +1,149 @@
+const { storeOrderRecipientSchema } = require('../schemas/store-order-recipient');
+
+const debugLogs = [];
+
+const setV5Token = async () => {
+  const {
+    data: { access_token: accessToken },
+  } = await axios.post(
+    `${config.get('piiServerUrl')}/api/v5/auth/token`,
+    {
+      grant_type: 'client_credentials',
+    },
+    {
+      auth: {
+        username: config.get('clientId'),
+        password: config.get('clientSecret'),
+      },
+    }
+  );
+  debugLogs.push({ function: 'storeOrderRecipient.getV5Token', accessToken });
+  axios.defaults.headers.common.Authorization = `Bearer ${accessToken}`;
+};
+
+const createOrder = ({ contextId }) => async (params, callback) => {
+  debugLogs.push({ function: 'createOrder', params });
+  try {
+    const response = await axios.post(`${config.get('piiServerUrl')}/api/v4/programs/${contextId}/orders`, params);
+    callback(null, response.data, response);
+  } catch (error) {
+    callback({ error, debugLogs });
+  }
+};
+
+/**
+ * Higher order function for creating a `create` order function in context.
+ *
+ * NOTE: This is a v5 API endpoint, therefore ignores the `apiVersion` config prop
+ *
+ * @param {module:resources/orders~OrderContext} orderContext Order context object
+ *
+ * @returns {module:resources/orders~CreateFunc} `create` order function in context
+ *
+ * @protected
+ */
+const storeOrderRecipient = ({ programCode }) => async member => {
+  await storeOrderRecipientSchema.validate(member).then(async () => {
+    try {
+      await setV5Token();
+    } catch (error) {
+      debugLogs.push({ function: 'storeOrderRecipient.getV5Token', error });
+      // eslint-disable-next-line no-throw-literal
+      throw { errors: { error, debugLogs } };
+    }
+
+    try {
+      const {
+        data: { result },
+      } = await axios.post(`${config.get('piiServerUrl')}/api/v5/programs/${programCode}/order_recipients`, member);
+      return result;
+    } catch (error) {
+      debugLogs.push({
+        function: 'storeOrderRecipient.createOrderRecipients',
+        error: get(error, 'response.data'),
+      });
+      // eslint-disable-next-line no-throw-literal
+      throw { errors: { error, debugLogs } };
+    }
+  });
+};
+
+const getOrderRecipient = ({ programCode }) => async (orderRecipientCode, callback) => {
+  try {
+    await setV5Token();
+    const { data } = await axios.get(
+      `${config.get('piiServerUrl')}/api/v5/programs/${programCode}/order_recipients/${orderRecipientCode}`
+    );
+    callback(null, data.result, data);
+  } catch (error) {
+    callback(error);
+  }
+};
+
+const orderRecipientFactory = (contextTypeName, contextId, programCode) => {
+  if (contextTypeName !== 'programs') {
+    throw new Error('Can only create an order recipient object for programs');
+  }
+  if (!config.get('piiServerUrl')) {
+    throw new Error('piiServerUrl is not configured');
+  }
+
+  const orderContext = {
+    contextTypeName,
+    contextId,
+    programCode,
```

Comment:
I think `contextId` and `programCode` are the same thing…

Reply:
From my understanding, contextId [integer] (or program id) is the v4 UUID, but v5 uses code [string] which is a separate column in their DB in an effort to conceal the DB id from the public.

I.e. demo company has an id of 252, but code 'demo_company'